src/context/cart-context.js

```diff
@@ -9,6 +11,35 @@ const CartContextProvider = ({ children }) => {
     cartList: [],
   });
 
+  const { cartList } = cartState;
+
+  useEffect(() => {
+    const encodedToken = localStorage.getItem("token");
+    const cartListLocalStorage = JSON.parse(localStorage.getItem("cartList"));
+    if (
+      cartList.length < 1 &&
+      cartListLocalStorage !== null &&
+      cartListLocalStorage.length >= 1
+    ) {
+      cartListLocalStorage.map((item) => {
+        return (async () => {
+          const response = await axios({
+            method: "post",
+            url: `/api/user/cart`,
+            headers: { authorization: encodedToken },
+            data: {
+              product: item,
```

Comment:
suggestion: I see this chunk of code is repeated many places, you can add some utility function for these lines as well so you can eliminate code duplication